docs/checks/overlappingAOIPolygonCheck.md

```diff
@@ -0,0 +1,178 @@
+# Overlapping AOI Polygon Check 
+
+This check flags polygons that overlap and represent the same Area of Interest (AOI).
+
+AOIs are defined through the configurable value `aoi.tags.filters`. This is a list of tag filters. If a polygon has any of the tags in any of the filters, it is considered an AOI.
+
+The defaults AOI tag filters are:  
+
+* `amenity->FESTIVAL_GROUNDS`
+* `amenity->GRAVE_YARD|landuse->CEMETERY`
+* `boundary->NATIONAL_PARK,PROTECTED_AREA|leisure->NATURE_RESERVE,PARK`
+* `historic->BATTLEFIELD`
+* `landuse->FOREST|natural->WOOD`
+* `landuse->RECREATION_GROUND|leisure->RECREATION_GROUND`
+* `landuse->VILLAGE_GREEN|leisure->PARK`
+* `leisure->GARDEN`
+* `leisure->GOLF_COURSE|sport->GOLF`
+* `leisure->PARK&name->*`
+* `natural->BEACH`
+* `tourism->ZOO`
+
+#### Live Examples
+
+1. The way [id:99881325](https://www.openstreetmap.org/way/99881325) overlaps way [id:173830769](https://www.openstreetmap.org/way/173830769) and they are both tagged with `leisure=PARK`.
+2. The way [id:54177792](https://www.openstreetmap.org/way/54177792), with tag `landuse=FOREST`, overlaps way [id:338963545](https://www.openstreetmap.org/way/338963545), with similar tag `natural=WOOD`.
+
+#### Code Review
+
+In [Atlas](https://github.com/osmlab/atlas), OSM elements are represented as Edges, Points, Lines, Nodes, Areas & Relations; in our case, we’re are looking at [Areas](https://github.com/osmlab/atlas/blob/dev/src/main/java/org/openstreetmap/atlas/geography/atlas/items/Area.java).
+
+First we check if the object is an Area and it has a tag that identifies it as an AOI.
+
+```java
+@Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Area && (aoiFiltersTest(object))
+                && !this.isFlagged(object.getIdentifier());
+    }
+```
+
+The AOI tag test is performed by looping through all the tag filters in the configurable value.
+
+```java
+private boolean aoiFiltersTest(final AtlasObject object)
+    {
+        for (final TaggableFilter filter : this.aoiFilters)
+        {
+            if (filter.test(object))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+```
+
+Then the Areas that overlap the initial Area are found, and tested to see if the minimum overlap is met (configurable) and if they have the same or similar AOI tag.
+
+```java
+@Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+
+        final Area aoi = (Area) object;
+        final Polygon aoiPolygon = aoi.asPolygon();
+        boolean hasOverlap = false;
+
+        // Set of overlapping area AOIs
+        final Set<Area> overlappingArea = Iterables.stream(object.getAtlas().areasIntersecting(
+                aoiPolygon.bounds(),
```

Comment:
I'm not too familiar with how this is setup. Two questions:
1. Why is the java code duplicated in the markdown document?
2. Should it be consistent with the changes made to the code below?

Reply:
Great question; the markdown docs are supposed to be a high level walkthrough, or tutorial, of the check code. This includes flag examples and a walkthrough, as well as detailed explanation of any complex/interesting functions.

In this case, there's probably more code copied over than needed 😃

Reply:
The markdown document is meant to give a higher level explanation of what the code is doing. Explanations of the code are paired with the code snippets to give context, that is why there is the duplication. As for the consistency, it should be the same. I will correct that.

Reply:
That makes sense - do we want to continue doing this? It sounds like the two can quickly get out of sync - as they are in this PR already. And looking at the last few, it's been the same code as in the actual implementation. It may be worthwhile updating the strategy around this. Just my 2 cents, you have more insight in this project!